src/kernels/types.ts

```diff
@@ -559,15 +559,20 @@ export enum InterruptResult {
 /**
  * Closely represents Jupyter Labs Kernel.IKernelConnection.
  */
-export interface IBaseKernelSession<T extends 'remoteJupyter' | 'localJupyter' | 'localRaw'> extends IAsyncDisposable {
-    readonly id?: string;
+export interface IBaseKernelSession<T extends 'remoteJupyter' | 'localJupyter' | 'localRaw'>
+    extends Session.ISessionConnection {
+    readonly id: string;
     readonly kind: T;
-    readonly disposed: boolean;
-    readonly kernel?: Kernel.IKernelConnection;
+    readonly isDisposed: boolean;
+    readonly kernel: Kernel.IKernelConnection | null;
```

Comment:
`null` not `undefined` in Jupyter lab npm packages.
Hence updated to line up with that